cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -116,6 +118,7 @@ public CaReconciler(
     ) {
         this.reconciliation = reconciliation;
         this.vertx = vertx;
+        this.workerExecutor = vertx.createSharedWorkerExecutor("kubernetes-ops-pool");
```

Comment:
I don't get this. Why is this needed? The reference is kept in `ClusterOperator` class. This should not create a new executor, it just gets an existing one. The same should apply to most of the other production classes apart from `ClusterOperator` and `UserOperator`

Reply:
This way the unit tests didn't need to change, like `KubernetesRestartEventsMockTest` has direct calls to the reconciler. I'm trying to avoid the failures occuring again in future because class X is used in a new unit test or context.

Reply:
Oh and `ZkImpl` uses a different pool, so that one needs a field.

Reply:
Alternatively we could make all the vertx tests use some common code to create and hold a worker executor?

Reply:
My personal preference would be to keep the production code clean of this and do it in the rest code. Not sure a common code can be used there, since each test will handle it differently.

Reply:
Changed it to create one field per operator, one field in `ZkImpl` and then do the rest in the test classes.